src/hooks/useDidUpdate/index.ts

```diff
@@ -0,0 +1,24 @@
+import {useEffect, useRef, useCallback} from 'react';
+import _ from 'lodash';
+
+/**
+ * This hook avoid calling useEffect on the initial value of his dependency array
+ */
+const useDidUpdate = (callback: () => void, dep: [any]) => {
+  const _dep = useRef<[any] | undefined>(dep);
+
+  const isDataChanged = useCallback(() => {
+    return !_.isEqual(_dep.current, dep);
+  }, [dep]);
```

Comment:
Not sure there's a need for useCallback here, you don't pass the function reference anywhere..

Reply:
Right, I should change it to useMemo to avoid re do the `_. isEqual` work